### ISSUE TITLE
is there a fix Error watching 'assets/' folder for changes: bad file descriptor?

### ISSUE THREAD

**Issue by maroza-training** [plentico/plenti#187]
Error watching 'assets/' folder for changes: bad file descriptor: any ideas on how to fix this?

**Comment by jimafisk** [comment]
@maroza-training if your project is public, can you link to it so I can take a look? Also, can you let me know what operating system you're on? 

Just taking a wild guess, but the build might break if you have spaces in the filenames of your assets instead of underscores.

**Comment by jimafisk** [comment]
I think I found you're repo: https://github.com/maroza-training/maroza-site. The site looks great by the way!

The current `main` branch is building ok on my Linux computer. Can you push up the breaking change so I can take a look? Thanks!

**Comment by jimafisk** [comment]
Potentially related:
- https://github.com/fsnotify/fsnotify/issues/51
- https://stackoverflow.com/questions/33851692/golang-bad-file-descriptor

**Comment by maroza-training** [comment]
so i'm using this site to learn plentico, web development and github, well add Go to that, plenti is wonderful, I think I love almost it.here in lies a problem with this error, if I directly clone the repo and do a local plenti serve, I get the error:

    On Thursday, April 7, 2022, 08:44:38 PM GMT+2, Jim Fisk ***@***.***> wrote:  
 
 


I think I found you're repo: https://github.com/maroza-training/maroza-site. The site looks great by the way!

The current main branch is building ok on my Linux computer. Can you push up the breaking change so I can take a look? Thanks!

**Comment by jimafisk** [comment]
@maroza-training that's great, I'm glad that we're part of your learning journey! It looks like you might have replied to this issue from your email and it possibly stripped out your error message, so I unfortunately can't see it. Can you try reposting it so I can take a look? Thanks!

**Comment by maroza-training** [comment]
maroza-site % plenti serve
███████▒▒▒ Building... Total build took 1.410284542s

Serving site from your "public" directory.
Visit your site at http://localhost:3000/
2022/04/07 21:48:40 errs.go:57: Error watching 'assets/' folder for changes: bad file descriptor
bad file descriptor (File: /home/runner/work/plenti/plenti/cmd/watcher.go, Line: 62)
2022/04/07 21:48:40 errs.go:57: couldn't add 'plenti.json' to watcher bad file descriptor
bad file descriptor (File: /home/runner/work/plenti/plenti/cmd/watcher.go, Line: 66)
2022/04/07 21:48:40 errs.go:57: couldn't add 'package.json' to watcher bad file descriptor
bad file descriptor (File: /home/runner/work/plenti/plenti/cmd/watcher.go, Line: 70)
2022/04/07 21:48:40 errs.go:57: Error watching 'assets/' folder for changes: bad file descriptor
bad file descriptor (File: /home/runner/work/plenti/plenti/cmd/watcher.go, Line: 62)
2022/04/07 21:48:40 errs.go:57: couldn't add 'plenti.json' to watcher bad file descriptor
bad file descriptor (File: /home/runner/work/plenti/plenti/cmd/watcher.go, Line: 66)
2022/04/07 21:48:40 errs.go:57: couldn't add 'package.json' to watcher bad file descriptor
bad file descriptor (File: /home/runner/work/plenti/plenti/cmd/watcher.go, Line: 70)

**Comment by jimafisk** [comment]
I think this is a Mac specific error, since your repository builds without error for me on Linux. I'll try to get my hands on a macbook in the next couple of days for some testing, in the meantime can you let me know which OS specifically you're using? Thanks!

**Comment by maroza-training** [comment]
yes it could be a Mac problem, I moved the build to a Linux machine and it works with issues

**Comment by maroza-training** [comment]
yes it could be a Mac problem, I moved the build to a Linux machine and it works without issues
